memo_app.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+require 'sinatra'
+require 'sinatra/reloader'
+require 'cgi'
+require 'json'
+require 'securerandom'
+
+MEMOS_FILE = 'memos.json'
+
+not_found do
+  'ページが見つかりません!'
+end
+
+def load_memos
+  JSON.parse(File.read(MEMOS_FILE), symbolize_names: true)
+end
+
+def save_memos(indicated_data)
+  File.write(MEMOS_FILE, JSON.pretty_generate(indicated_data))
+end
+
+def find_memo(existing_data)
+  existing_data.find { |memo| memo[:id] == params[:id] }
+end
+
+def assign_id
+  SecureRandom.uuid
+end
+
+get '/' do
+  @memos = load_memos
+  erb :index
+end
+
+get '/new' do
+  erb :new
+end
+
+post '/create' do
+  existing_data = load_memos
+  existing_data << { id: assign_id, title: params[:title], content: params[:content], created_at: Time.now, updated_at: Time.now }
+  save_memos(existing_data)
```

Comment:
```suggestion
  add_memos(params)
```

こんな感じで新規追加専用のメソッドを作って、そこにparamsを渡せばいい感じに保存してくれるメソッドを作ってみるといいかなと思います。

Reply:
ルーテイングには、メソッドを呼び出すアクションを置き、処理内容はメソッドに任せる。こちらの方が見やすいですね！